cpuusage beacon

NikulovE opened the pull request:
### What does this PR do?
CPU Usage Beacons which will available for Windows and for Unix
### What issues does this PR fix or reference?
no info
### Tests written?
salt win2008 beacons.add cpuusage "{'percent': '28'}"
Works perfectly

beacons.load is not available for Windows.

[cpuusage.txt](https://github.com/saltstack/salt/files/901522/cpuusage.txt)

terminalmage committed:
Don't use context caching for gathering systemd services

In a single Salt run, when multiple pkg states (with accompanying
service states) are run, the first service state establishes
__context__['systemd.systemd_services']. Then, subsequent pkg states
which add new packages will in effect add new systemd services, but
systemd.py is oblivious to this change and further attempts to get the
list of systemd services are returned from the (now outdated) cache.

This commit resolves this by not using context caching for the list of
systemd services.

Commit:
Remove unused import for lint

terminalmage committed:
systemd.py: when getting all services, don't repeat gathering of systemd services

Now that we are not caching these in the context dunder, we should pass
them into this function so that we don't repeat gathering the systemd
services inside _get_sysv_services().

Also, this fixes an incorrect change to the mocking that was apparently
made at some point to the get_all unit test.

terminalmage committed:
Merge branch 'issue40279' of https://github.com/terminalmage/salt into issue40279

terminalmage committed:
Backport auth and custom registry fixes from #40480 to 2016.3 branch

terminalmage committed:
Make sure we keep the cached client when clearing context

gtmanfred committed:
update docs for logging handlers

Commit:
Merge pull request #40505 from gtmanfred/2016.3

update docs for logging handlers

Commit:
Remove directory content instead of directory itself when using `force_clone` in `git.latest` state. (refs #31363)

Commit:
Changed rm_rf's argument to actually remove intended file. (refs #31363)

Commit:
Merge pull request #40481 from terminalmage/docker-auth-handling-2016.3

Backport auth and custom registry fixes from #40480 to 2016.3 branch

Commit:
Merge pull request #40306 from terminalmage/issue40279

Don't use context caching for gathering systemd services

terminalmage committed:
Check master's ssh_minion_opts for fileserver/pillar values and ignore them

This prevents the ssh_minion_opts from overwriting values that should
have been inherited or otherwise setup by the master for the SSH minion.

Resolves #39892

Commit:
Merge pull request #40534 from terminalmage/issue39892

Check master's ssh_minion_opts for fileserver/pillar values and ignore them

Ch3LL committed:
fix vultr cloud race condition to match on 0*

Ch3LL committed:
check for salt install fail on vultur test

terminalmage committed:
Merge branch 'fix-31363' of https://github.com/zer0def/salt into zer0def-fix-31363

terminalmage committed:
Merge branch 'zer0def-fix-31363' into 2016.3

terminalmage committed:
Don't mark files that already were deleted as errors

Also use six.iteritems()

terminalmage committed:
Don't use __opts__.get() for hash_type

First of all, the opts will never not have this key, as it has a default
value in salt/config/__init__.py. Second, the default hash type changed
to sha256 anyway in 2016.11, so we shouldn't be referring to a specific
hash type when it'll no longer be accurate after a merge-forward.

salt-jenkins commented:
@g-shockfx, thanks for your PR! By analyzing the history of the files in this pull request, we identified @terminalmage, @ffa and @jacksontj to be potential reviewers.

Commit:
Merge pull request #40548 from Ch3LL/fix_vultrpy

Fix vultrpy

terminalmage committed:
Fix dockerng _get_client() regression

This fixes a regression I introduced the other day in #40481 when I
backported auth fixes to 2016.3. I changed how the client was
instantiated and ended up passing an unsupported kwarg to the wrapped
function. This resolves that regression.

cachedout commented:
Hi @g-shockfx 

It looks like you wrote this code based on the 2016.3 branch but your PR submitted it against the develop branch. We need this code based off the develop branch so we need you to please rebase your commits against that branch. Let us know if you need any help doing that. Thanks!

terminalmage committed:
Allow pillar.get to merge list as well as dictionaries

This also gets rid of the exception that used to be raised when the
default was not of the proper type, and instead skips merging in those
cases.

Commit:
Merge pull request #40562 from terminalmage/fix-get-client

Fix dockerng _get_client() regression

Commit:
Merge pull request #40552 from terminalmage/fix-hash-type-refs

Don't use __opts__.get() for hash_type

Commit:
Merge pull request #40567 from terminalmage/fix-pillar-get-merge-lists

Allow pillar.get to merge list as well as dictionaries

cachedout commented:
@g-shockfx If you're going to change the base branch, it should be changed to 2016.3, not to develop.
